src/Task/PhpCsFixerV2.php

```diff
@@ -72,7 +72,15 @@ public function run(ContextInterface $context)
         $arguments->addOptionalArgument('--allow-risky=%s', $config['allow_risky'] ? 'yes' : 'no');
         $arguments->addOptionalArgument('--cache-file=%s', $config['cache_file']);
         $arguments->addOptionalArgument('--config=%s', $config['config']);
-        $arguments->addOptionalCommaSeparatedArgument('--rules=%s', $config['rules']);
+
+        $rules = $config['rules'];
+        // JSON-encode rules if specified as a map.
+        if (array_values($rules) !== $rules) {
+            $arguments->addOptionalArgument('--rules=%s', json_encode($rules));
+        } else {
```

Comment:
I think the easiest thing to do is something like this:

```php
        if (count($rules)) {
            $arguments->add(sprintf(
                '--rules=%s', 
                array_values($rules) === $rules ? implode(',', $rules) : json_encode($rules)
            ));
        }
```

The current implementation of the json part will always add an empty array. Even though there are no rules configured. Since this is a pretty custom argument, it is OK not to use the `ProcessArgumentsCollection`.